lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -34,6 +34,52 @@ def self.load_db_adapter!
       require_relative("active_record/#{PolicyElement.configurations[Rails.env]['adapter']}")
     end
 
+    def self.can_buffer?
+      respond_to?(:persist_buffers!)
+    end
+
+    def self.buffering?
+      @buffering
+    end
+
+    def buffering?
+      self.class.buffering?
+    end
+
+    def self.start_buffering!
+      @buffering = true
+    end
+
+    def self.stop_buffering!
+      @buffering = false
+    end
+
+    def self.buffers
+      #The associations thing being different is annoying. Causality sucks.
+      @buffers ||= {upsert: {}, delete:{}, assignments: {}, associations: [] }
+    end
+
+    def self.clear_buffers!
+      @buffers = nil
+    end
+
+    def buffers
+      self.class.buffers
+    end
+
+    def self.persist_buffers!
+      # TODO: if postgres needs all conflict targets specified, do we need this to be injectable?
+      # Also, we must move this into the pg adapter, since mysql behaves differently
+      # https://github.com/zdennis/activerecord-import/wiki/On-Duplicate-Key-Update
```

Comment:
I haven't tested it, but does mysql actually break here? I assumed passing it a truthy value would be enough and it'd just ignore the fact that it's an array.